src/proxy/offline/proxy.js

```diff
@@ -304,24 +304,22 @@ export default _.defaults(
                         .flush()
                         .then(function(data) {
                             actions = data;
+                            self.syncInProgress = false;
```

Comment:
this should be move after the promise is resolved, nor before

Reply:
something like:
```javascript
return this.queue.serie(function() {
                    return self.actionStore
                        .flush()
                        .then(function(data) {
                            actions = data;
                            if (data && data.length) {
                                return self.sendSyncData(data);
                            }
                        })
                        .catch(function(err) {
                            if (self.isConnectivityError(err)) {
                                self.setOffline('communicator');
                                _.forEach(actions, function(action) {
                                    self.actionStore.push(action.action, action.parameters, action.timestamp);
                                });
                                return;
                            }

                            self.syncInProgress = false;
                            self.trigger('error', err);

                            throw err;
                        })
                        .then(data => {
                            self.syncInProgress = false;
                            return data;
                        } );
                });
```

Reply:
fixed